x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/transport/SecurityServerTransportInterceptor.java

```diff
@@ -63,6 +65,19 @@ public SecurityServerTransportInterceptor(
         SSLService sslService,
         SecurityContext securityContext,
         DestructiveOperations destructiveOperations
+    ) {
+        this(settings, threadPool, authcService, authzService, sslService, securityContext, destructiveOperations, null);
+    }
+
+    public SecurityServerTransportInterceptor(
+        Settings settings,
+        ThreadPool threadPool,
+        AuthenticationService authcService,
+        AuthorizationService authzService,
+        SSLService sslService,
+        SecurityContext securityContext,
+        DestructiveOperations destructiveOperations,
+        Client client
```

Comment:
If we go for implementing this, we could expose just a cluster alias resolver which uses `client.getRemoteClusterAliasForConnection()` under the hood instead of injecting the whole client instance.